src/runtime/bfile.c

```diff
@@ -52,6 +60,37 @@ bfbuffer_snoc(struct bfbuffer *bf, int byte)
   bf->buf[bf->pos++] = byte;
 }
 
+size_t
+bfbuffer_read(struct bfbuffer *bf, uint8_t *buf, size_t size)
+{
+  if (bf->pos + size > bf->size) {
```

Comment:
Can't you just say `if (bf->pos + size > bf->size) { size = bf->size - bf->pos; }` ?

Reply:
Yes, that would also work.